### PR TITLE
go 1.7

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Comment by ilovezfs** [comment]
godep needs a revision bump. other go build tools may need one too.

**Comment by DomT4** [comment]
`godep` seems to be working fine locally from the old bottle. Is there an issue I'm not remembering here? May well be, to be fair.

**Comment by DomT4** [comment]
Oh, I see. Found the prior discussion. Let's bump anyway, they're tiny tools & it won't hurt.

File: Formula/go.rb [hunk 1]
```diff
@@ -3,27 +3,16 @@ class Go < Formula
...
-      version "1.6.2"
-      sha256 "6ebbafcac53bbbf8c4105fa84b63cca3d6ce04370f5a04ac2ac065782397fc26"
-    end
+    url "https://storage.googleapis.com/golang/go1.7.src.tar.gz"
+    mirror "https://fossies.org/linux/misc/go1.7.src.tar.gz"
+    version "1.7"
+    sha256 "72680c16ba0891fcf2ccf46d0f809e4ecf47bbf889f5d884ccb54c5e9a17e1c0"
 
-    go_version = "1.6"
+    go_version = "1.7"
```
Comment (ilovezfs) [inline Formula/go.rb]: would prefer go_version = version.to_s.split(".")[0..1].join(".") if the goal is to use "1.7" for "1.7", "1.7.1", "1.7.2", ...

File: Formula/go.rb [hunk 2]
```diff
@@ -101,7 +67,7 @@ def install
       system "#{bin}/go", "install", "-race", "std"
     end
 
-    if build.with?("godoc") || build.with?("vet")
+    if build.with?("godoc")
```
Comment (ilovezfs) [inline Formula/go.rb]: I think you can drop the parentheses now, no?

Comment (DomT4) [inline Formula/go.rb]: Yeah. Not my change here, I overlooked it.

File: Formula/go.rb (hunk 1 above)
Comment (DomT4) [inline Formula/go.rb]: Meh, let's not sex it up unless it proves necessary.

**Comment by DomT4** [comment]
Lord am I glad I was anal enough to run around fixing these the other week:

> /usr/local/Cellar/go/1.7/libexec/pkg/tool/darwin_amd64/link: -X flag requires argument of the form importpath.name=value

**Comment by ilovezfs** [comment]
indeed.

**Comment by ilovezfs** [comment]
was fugu fail reported upstream?

File: Formula/go.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/go.rb]: already use that elsewhere. would prefer not to update this every time.

**Comment by DomT4** [comment]
Not yet, but have been watching it for a while. No activity in a year, and the last stable version of Go they tested against was Go _1.4_. We're not going to get a quick resolution there, but I will file it.

File: Formula/go.rb (hunk 1 above)
Comment (DomT4) [inline Formula/go.rb]: Alright, I'll do it just for you. Don't say I'm never nice.

Comment (ilovezfs) [inline Formula/go.rb]: thanks. we can kill it next time if it proves to be wrong.

File: Formula/godep.rb [hunk 3]
```diff
@@ -23,16 +23,20 @@ def install
...
+            "ImportPath": "go.googlesource.com/tools",
+            "Rev": "3fe2afc9e626f32e91aff6eddb78b14743446865"
+          }
+        ]
       }
     EOS
-    (testpath/"src/foo/bar/Godeps").install "Godeps.json"
-    cd("src/foo/bar") { system bin/"godep", "path" }
+    system bin/"godep", "restore"
+    assert File.exist?("src/go.googlesource.com/tools/README")
```
Comment (ilovezfs) [inline Formula/godep.rb]: I think we've been adding a message following these assert exists

File: Formula/go.rb [hunk 4]
```diff
@@ -34,51 +23,28 @@ class Go < Formula
...
-    end
-
     resource "gotools" do
       url "https://go.googlesource.com/tools.git"
     end
   end
 
   option "without-cgo", "Build without cgo"
   option "without-godoc", "godoc will not be installed for you"
-  option "without-vet", "vet will not be installed for you"
```
Comment (ilovezfs) [inline Formula/go.rb]: I think removing this is going to force source builds for everyone who currently has `without-vet` in their INSTALL_RECEIPT but that's not your fault.

File: Formula/go.rb [hunk 5]
```diff
@@ -34,51 +23,28 @@ class Go < Formula
...
   option "without-race", "Build without race detector"
 
   depends_on :macos => :mountain_lion
 
+  # Should use the last stable binary release to bootstrap.
+  # More explicitly, leave this at 1.7 when 1.7.1 is released.
```
Comment (ilovezfs) [inline Formula/go.rb]: Is there a citation for this? Insisting on stage 1 not stage 2 bootstraps sounds really weird to me.

**Comment by ilovezfs** [comment]
Which formulae has this PR been tested with already?

File: Formula/go.rb (hunk 4 above)
Comment (DomT4) [inline Formula/go.rb]: Not sure, but yeah, can't be avoided here 😢

File: Formula/go.rb (hunk 5 above)
Comment (DomT4) [inline Formula/go.rb]: Can you elaborate on which part you want a citation for?

File: Formula/godep.rb (hunk 3 above)
Comment (DomT4) [inline Formula/godep.rb]: Trying to get this done fairly quickly; also working on GPG stuff in the background which is more important. Just wanted to push this through ASAP so I didn't wake up to 450 thumbs up in the other thread later.

**Comment by ilovezfs** [comment]
Does this break gox? I'm wondering if the formulae that bootstrap gox first from go_resource blocks may need updates to a newer gox revision.

File: Formula/go.rb (hunk 5 above)
Comment (ilovezfs) [inline Formula/go.rb]: why would using 1.7.1 for bootstrap be wrong (for next update of course not this one since 1.7.1 doesn't exist yet)

Comment (DomT4) [inline Formula/go.rb]: We had this discussion recently 😉. It enables us to bootstrap with a build we know with a high degree of certainty works, so if there's build errors we know the errors are likely with the new release. Also gets people off our back around wasting their lives by building from source blah blah blah.

**Comment by ilovezfs** [comment]
Anyway, regardless of comments, this looks great. I'd be in favor of shipping as soon as you have :green_apple:

**Comment by DomT4** [comment]
Typically there have never been compatibility problems with `gox` on major updates. This seems to be the case again this time from a cursory check.

File: Formula/go.rb (hunk 5 above)
Comment (ilovezfs) [inline Formula/go.rb]: does upstream not do stage 2 builds?

**Comment by ilovezfs** [comment]
Yeah, I'll probably trigger a rebuild of all go everything locally after this ships but I'm expecting very little breakage if any.

**Comment by DomT4** [comment]
I did a partial rebuild locally & it didn't blow up. I think Homebrew's exposure to 1.7 breakage is fairly low, compared particularly to previous major releases. Always happy to be unpleasantly surprised though 🙈.

**Comment by DomT4** [comment]
Shipped!

**Comment by ilovezfs** [comment]
hurray! :tada:

**Pull request closed without merging**